include/DeterministicNoiseSimulator.hpp

```diff
@@ -43,83 +71,56 @@ class DeterministicNoiseSimulator: public Simulator {
 
     std::map<std::string, double> DeterministicSimulate();
 
-    //    std::map<std::string, std::string> AdditionalStatistics() override {
-    //        return {
-    //                {"step_fidelity", std::to_string(step_fidelity)},
-    //                {"approximation_runs", std::to_string(approximation_runs)},
-    //                {"final_fidelity", std::to_string(final_fidelity)},
-    //        };
-    //    };
-
     [[nodiscard]] std::string intToString(long target_number, char value) const;
 
     void applyDetNoiseSequential(const qc::Targets& targets);
 
-    [[nodiscard]] std::map<std::string, double> AnalyseState(dd::QubitCount nr_qubits, bool full_state) const;
+    [[nodiscard]] std::map<std::string, double> AnalyseState(dd::QubitCount nr_qubits, bool full_state);
 
-    void setNoiseEffects(const std::string& cGateNoise) { gateNoiseTypes = cGateNoise; }
+    void setNoiseEffects(const std::string& cGateNoise) {
+        for (const auto& effect: cGateNoise) {
+            if (effect != 'A' && effect != 'P' && effect != 'D') {
+                throw std::runtime_error("Unknown noise operation '" + std::to_string(effect) + "'\n");
+            }
+        }
```

Comment:
```suggestion
        if (cGateNoise.find_first_not_of("APD") != std::string::npos) {
            throw std::runtime_error("Unknown noise operation in '" + cGateNoise + "'\n");
        }
```

Slightly more idiomatic C++ :)

Is this missing `D`epolarizing?